src/wp-includes/blocks/index.php

```diff
@@ -12,6 +12,54 @@
 require BLOCKS_PATH . 'widget-group.php';
 require BLOCKS_PATH . 'require-dynamic-blocks.php';
 
+/**
+ * Registers core block types styles.
+ *
+ * @since 6.3.0
+ */
+function register_core_block_types_styles() {
+	static $core_blocks_meta;
+	if ( ! $core_blocks_meta ) {
+		$core_blocks_meta = require ABSPATH . WPINC . '/blocks/blocks-json.php';
+	}
+
+	$suffix = SCRIPT_DEBUG ? '' : '.min';
+
+	$style_fields = array(
+		'editorStyle',
+		'style',
+	);
+
+	foreach ( $core_blocks_meta as $name => $schema ) {
+		foreach ( $style_fields as $style_field ) {
+			if ( ! isset( $schema[ $style_field ] ) ) {
+				$style_handle = 'style' === $style_field ? "wp-block-{$name}" : "wp-block-{$name}-editor";
+				wp_register_style(
+					$style_handle,
+					false
+				);
+				continue;
+			}
+			$style_handle = $schema[ $style_field ];
+			if ( is_array( $style_handle ) ) {
+				continue;
+			}
+			$path = "/wp-includes/blocks/$name/style{$suffix}.css";
+			wp_register_style( $style_handle, $path );
```

Comment:
This needs to use the URL like in `register_block_style_handle()` instead of path:
```suggestion
			wp_register_style( $style_handle, includes_url( "blocks/$name/style{$suffix}.css" );
```

Reply:
If you look at `wp_default_scripts`, other core styles do not do this. I am not sure why, but they do not. It is worth noting that most of these styles will never use this url for anything, as they will be inlines by `wp_maybe_inline_styles`. 

I do see why this should different from `wp_default_scripts`, but I could be talked out of it.

Reply:
Better to keep this like it was before / like it is in `register_block_style_handle()`. I see you updated it, so all good now 👍